### PR TITLE
Improvements for handling CodeChecker version checks

### LINKED ISSUE #111: "CodeChecker not found" and "Found supported CodeChecker" popups on every reconnect
Every time I reconnect to my workspace (Reload Window, VPN disconnect, etc.) with a source file open, I get the following popups from CodeChecker:
image

Here's the output from the extension when this occurs:
```
>>> Database found at path: /depot/out/cur/compile_commands.json
{"base_package_version": "6.19.1", "package_build_date": "2022-03-18T16:12", "git_commit": "9fa0f9817f6e91eda2a9202e0a3bf5006accdc8c", "git_tag": "6.19.1"}
>>> Process 'CodeChecker analyzer-version' exited with code 0
>>> Supported CodeChecker version 6,19,1, enabled
>>> Starting process 'CodeChecker parse'
> CodeChecker parse /depot/out/cur/codechecker/reports -e json --file /depot/.../source.cc
>>> Database found at path: /depot/out/cur/compile_commands.json
>>> Process killed
>>> Starting process 'CodeChecker parse'
> CodeChecker parse /depot/out/cur/codechecker/reports -e json --file /depot/.../source.cc
>>> Database found at path: /depot/out/cur/compile_commands.json
>>> Database found at path: /depot/out/cur/compile_commands.json
>>> Process 'CodeChecker parse' exited with code 0
```

This doesn't happen if I close all editors and then reload the window, but it seems that any file being open during the reload will trigger it, it doesn't necessarily need to be a source file I'm trying to analyze.

So there's kind of three issues:
1. Why is it reporting that it couldn't find it but then could?
2. Can we get an option to suppress the "Found supported CodeChecker" popup all the time?
3. Why is parsing happening twice without me taking any action during startup? Is it because of the `Process killed` line? What killed it?

My config (in my `.code-workspace` file):
```
{
    "codechecker.backend.compilationDatabasePath": "/depot/out/cur/compile_commands.json",
    "codechecker.backend.outputFolder": "/depot/out/cur/codechecker",
    "codechecker.executor.arguments": "--skip /depot/build/config/codechecker.skipfile --ctu --analyzer-config 'clang-tidy:take-config-from-directory=true'",
}
```

### PR DESCRIPTION
**Pull request by Discookie**
Fixes #111.

* Refactored the `CodeChecker version` handling, so a version check process couldn't be prematurely killed, leading to a false `CodeChecker not found` notification. (Should be enough to fix the particular issue showing double notifs.)
* Added a toggle to disable version-check related notifications altogether

### CODE REVIEW

File: src/editor/executor.ts [hunk 1]
```diff
@@ -110,7 +110,13 @@ export class ExecutorAlerts {
             break;
         case ProcessStatus.errored:
             this.statusBarItem.text = '$(testing-error-icon) CodeChecker: analysis errored';
-            window.showErrorMessage('CodeChecker finished with error - see logs for details');
+
+            const shouldShowError = workspace.getConfiguration('codechecker.executor')
```
Comment (vodorok) [inline src/editor/executor.ts:114]: I wouldn't hide Codechecker runtime error notifications in the case when the user ticks the newly introduced option. The settings description says: "Enable notifications when CodeChecker is not found or not supported", but I suspect there can be other errors too. Partial/total suppression of notifications could be introduced, like with a slider or with another tickbox, that says, "Disable every popup originating from the plugin" or something along the way.

If the cases for a failed analysis and other runtime errors are handled elsewhere than this, please consider the above remarks irrelevant.

**Review by vodorok — CHANGES_REQUESTED** [review]
Please see my only remark, otherwise good job!

File: src/editor/executor.ts (hunk 1 above)
Comment (Discookie) [reply src/editor/executor.ts]: Changed the option to `Enable CodeChecker-related notifications`, as other notifications being spammed has also been an issue before. It now controls every notification, except when a report cannot be found when trying to jump to it.

I'm planning to show the notification's contents in the sidebar, but that will be added in another PR.

File: src/editor/initialize.ts [hunk 2]
```diff
@@ -20,6 +21,10 @@ export class FolderInitializer {
     }
 
     async showDialog() {
+        if (shouldShowNotifications()) {
```
Comment (vodorok) [inline src/editor/initialize.ts:24]: Early return if notifications are enabled? Shouldn't the condition be negated?

**Review by vodorok — CHANGES_REQUESTED** [review]
Please see my remark

File: src/editor/initialize.ts (hunk 2 above)
Comment (Discookie) [reply src/editor/initialize.ts]: Whoops, fixed.

**Review by vodorok — APPROVED** [review]
LGTM

**Pull request merged**